### PR TITLE
colorful-boxes:0.1.0

### PR DESCRIPTION
**Pull request by lkoehl**
I am submitting
- [x] a new package
- [ ] an update for a package

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name in conformance with the guidelines
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] ensured that my submission does not infringe upon the rights of a third party
- [x] tested my package locally on my system and it worked
- [x] named this PR as `name:version` of the submitted package
- [x] agree that my package will not be removed without good reason

Description: This pull request introduces the 'colorful-boxes' package, an exciting addition to our repository that brings a collection of predefined and prestyled boxes, making it effortless to create documents with modern designs.

### CODE REVIEW

**Comment by reknih** [comment]
Hey and thank you for the submission. I have rejected the package name on the basis that it it still merely descriptive of what the package will do (draw colorful boxes). I'm aware of the discussion in lkoehl/typst-boxes#2 and am sorry that this project needs yet another name! The examples look very nice and I'd be glad to get this merged.

**Comment by lkoehl** [comment]
@reknih would it be acceptable if I rename the package to `boxtopia`? 

You probably had some good reasons to not allow, but I want to raise my concern that descriptive names could reduce confusion when having multiple packages for the same kind of topic (e.g. boxes) when all the names revolve around "box" but don't further indicate what they do. It's kind of the thing when naming variables.

```
#import "@preview/colorful-boxes:0.1.0": *
```

makes far more sense than

```
#import "@preview/boxtopia:0.1.0": *
```

**Comment by reknih** [comment]
Boxtopia would work!

Let me review this with Laurenz. I'll be able to give you more feedback tomorrow.

**Comment by reknih** [comment]
We have reconsidered and will allow this name. It still leaves the possibility for other, similar packages, without being too general (e.g. `fancy-boxes`, `colorful-containers`, ...).

File: packages/preview/colorful-boxes-0.1.0/typst.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "colorful-boxes"
+version = "0.1.0"
+repository = "https://github.com/lkoehl/typst-boxes"
+entrypoint = "lib.typ"
+authors = ["Lukas Köhl <https://github.com/lkoehl>"]
+license = "MIT"
+description = "Predefinded colorful boxes"
```
Comment (laurmaedje) [inline packages/preview/colorful-boxes-0.1.0/typst.toml:8]:
```suggestion
description = "Predefined colorful boxes"
```

File: packages/preview/colorful-boxes-0.1.0/lib.typ [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        (0pt, size.height + (2*inset)),
+        (size.width + (2*inset), size.height + (2*inset)),
+        (size.width + (2*inset) + 6pt, 0cm)
+      )
+      #place(center + top, dy: size.height, dx: -3pt)[#body]
+    ]]
+  })
+}
+
+#let slantedColorbox(title: "title", color: none, radius: 0pt, width: auto, body) = {
```
Comment (laurmaedje) [inline packages/preview/colorful-boxes-0.1.0/lib.typ:57]: Just wanted to mention that the canonical Typst case would be `slanted-colorbox`, but ultimately it's up to you.

**Review by laurmaedje — COMMENTED** [review]
I left two small comments, I'm sorry about the slow process.

**Comment by laurmaedje** [comment]
Just a heads up: The package directory layout changed. It's now `packages/preview/package/version` instead of `packages/preview/package-version`.

**Comment by lkoehl** [comment]
Hm, a rebase wasn't the smartest idea. Will create a new pull request

**Pull request closed without merging**